protocol/x/subaccounts/keeper/negative_tnc_subaccount.go

```diff
@@ -0,0 +1,62 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gogotypes "github.com/cosmos/gogoproto/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+// GetNegativeTncSubaccountSeenAtBlock gets the last block height a negative TNC subaccount was
+// seen in state. Note this defaults to 0 if it has not been set.
+func (k Keeper) GetNegativeTncSubaccountSeenAtBlock(
+	ctx sdk.Context,
+) uint32 {
+	store := ctx.KVStore(k.storeKey)
+	return k.getNegativeTncSubaccountSeenAtBlock(store)
+}
+
+// getNegativeTncSubaccountSeenAtBlock is a helper function that takes a store and returns the last
+// block height a negative TNC subaccount was seen in state.
+func (k Keeper) getNegativeTncSubaccountSeenAtBlock(
+	store sdk.KVStore,
+) uint32 {
+	b := store.Get(
+		[]byte(types.NegativeTncSubaccountSeenAtBlockKey),
+	)
+	blockHeight := gogotypes.UInt32Value{Value: 0}
```

Comment:
I almost wonder if we should be pessimistic here and return a non zero default value 🤔

Reply:
Honestly I prefer returning 0, since then it won't cause issues when re-starting the network from genesis (although I'll admit the current value will gate withdrawals in the first 50 blocks after re-starting from genesis, which I personally didn't think was an issue)